### ISSUE TITLE
gdb: No struct type named runtime.rtype

### ISSUE THREAD

**Issue by andredasilvapinto** [golang/go#9326]
I'm having an error while trying to manually load runtime-gdb.py on GDB.

go1.4 linux/amd64
gdb 7.8
Ubuntu 14.04.1 LTS

```
gdb ./collect-jmx.test -d $GOROOT
...
(gdb) source /usr/local/go/src/runtime/runtime-gdb.py 
Loading Go Runtime support.
Traceback (most recent call last):
  File "/usr/local/go/src/runtime/runtime-gdb.py", line 205, in <module>
    _rctp_type = gdb.lookup_type("struct runtime.rtype").pointer()
gdb.error: No struct type named runtime.rtype.
```

**Comment by pnasrat** [comment]
Change out for review

https://go-review.googlesource.com/#/c/2821/

**Comment by zhengzhiren** [comment]
I have the same issue.
CentOS 6.5
gdb 7.9
golang 1.4.2

```
(gdb) source ~/go/src/runtime/runtime-gdb.py
Loading Go Runtime support.
Traceback (most recent call last):
  File "~/go/src/runtime/runtime-gdb.py", line 205, in <module>
gdb.error: No struct type named runtime.rtype.
```

**Comment by cevaris** [comment]
+1
```
(gdb) source /go/src/runtime/runtime-gdb.py
Loading Go Runtime support.
Traceback (most recent call last):
  File "/go/src/runtime/runtime-gdb.py", line 205, in <module>
    _rctp_type = gdb.lookup_type("struct runtime.rtype").pointer()
gdb.error: No struct type named runtime.rtype.
(gdb)
```

**Comment by qingfeng-fy** [comment]
you shold modify runtime-gdb.py:
205 _rctp_type = gdb.lookup_type("struct runtime.rtype").pointer()
206
207
208 def iface_commontype(obj):
209         if is_iface(obj):
210                 go_type_ptr = obj['tab']['_type']
211         elif is_eface(obj):
212                 go_type_ptr = obj['_type']
213         else:
214                 return
215
216        return go_type_ptr.cast(_rctp_type).dereference()

to:
205 #_rctp_type = gdb.lookup_type("struct runtime.rtype").pointer()
206
207
208 def iface_commontype(obj):
209         if is_iface(obj):
210                 go_type_ptr = obj['tab']['_type']
211         elif is_eface(obj):
212                 go_type_ptr = obj['_type']
213         else:
214                 return
215
216 #       return go_type_ptr.cast(_rctp_type).dereference()
217         return go_type_ptr.cast(gdb.lookup_type("struct runtime.rtype").pointer()).dereference()

**Comment by anacrolix** [comment]
@qingfeng-fy Thanks for the suggestion, I recommend a diff next time. Unfortunately this only moves the error to later:

(gdb) info goroutine
Python Exception <class 'gdb.error'> There is no member named status.: 
Error occurred in Python command: There is no member named status.

(gdb) goroutine 1
Python Exception <class 'ValueError'> need more than 1 value to unpack: 
Error occurred in Python command: need more than 1 value to unpack

This makes it difficult to use gdb in addressing #5838

**Comment by minux** [comment]
This issue has been fixed on tip.

Please verify that you're using tip.

**Comment by anacrolix** [comment]
I will give it a try, thank you. Have you considered backporting the fix to 1.4.3?

**Comment by minux** [comment]
I'm not sure if/when Go 1.4.3 will happen.
but I do know that this issue doesn't meet
the bar to be included into a minor release.
(only serious bugs without workarounds
qualify.)

**Comment by anacrolix** [comment]
It works for me on tip.

**Comment by huangminlong** [comment]
@anacrolix, i meet the same problem with you.  i have run the stringutil.test in my platform and modify the runtime-gdb.py . who can tell me how to solve it?

Linux/CentOS 7.1
golang 1.4.2
gdb GNU gdb (GDB) Red Hat Enterprise Linux 7.6.1-64.el7

**Comment by huangminlong** [comment]
@anacrolix, i meet the same problem with you.  i have run the stringutil.test in my platform and modify the runtime-gdb.py . who can tell me how to solve it?

Linux/CentOS 7.1
golang 1.4.2
gdb GNU gdb (GDB) Red Hat Enterprise Linux 7.6.1-64.el7

**Comment by huangminlong** [comment]
@anacrolix, i meet the same problem with you.  i have run the stringutil.test in my platform and modify the runtime-gdb.py . who can tell me how to solve it?

Linux/CentOS 7.1
golang 1.4.2
gdb GNU gdb (GDB) Red Hat Enterprise Linux 7.6.1-64.el7

**Comment by davecheney** [comment]
@huangminlong this issue is fixed in Go 1.5 or later. Please open a new issue if you are still experiencing problems after upgrading to Go 1.5
